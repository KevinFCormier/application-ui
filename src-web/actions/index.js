/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/*
 * action types
 *
 * action name: <NOUN>_<VERB>
 */

export const CATALOG_DROPDOWN_FILTERS_VISIBILITY_TOGGLE = 'CATALOG_DROPDOWN_FILTER_VISIBILITY_TOGGLE'
export const CATALOG_FETCH_ERROR_STATUS_CHANGE = 'CATALOG_FETCH_ERROR_STATUS_CHANGE'
export const CATALOG_INSTALL_FAILURE = 'CATALOG_INSTALL_FAILURE'
export const CATALOG_INSTALL_LOADING = 'CATALOG_INSTALL_LOADING'
export const CATALOG_INSTALL_SUCCESS = 'CATALOG_INSTALL_SUCCESS'
export const CATALOG_INSTALL_VALIDATION_FAILURE = 'CATALOG_INSTALL_VALIDATION_FAILURE'
export const CATALOG_RESOURCE_FILTER_REPOS = 'CATALOG_RESOURCE_FILTER_REPOS'
export const CATALOG_RESOURCE_FILTER_SEARCH = 'CATALOG_RESOURCE_FILTER_SEARCH'
export const CATALOG_RESOURCE_SELECT = 'CATALOG_RESOURCE_SELECT' // clicking on a tile

export const CLEAR_REQUEST_STATUS = 'CLEAR_REQUEST_STATUS'

export const CREATE_GRAFANA_DASHBOARD = 'CREATE_GRAFANA_DASHBOARD'

export const DEL_RECEIVE_FAILURE = 'DEL_RECEIVE_FAILURE'
export const DEL_RECEIVE_SUCCESS = 'DEL_RECEIVE_SUCCESS'
export const DEL_REQUEST = 'DEL_REQUEST'

export const LOGS_RESET= 'LOGS_RESET'
export const LOGS_SEARCH = 'LOGS_SEARCH'

export const MODAL_UPDATE = 'MODAL_UPDATE'

export const NAMESPACES_RECEIVE_FAILURE = 'NAMESPACES_RECEIVE_FAILURE'
export const NAMESPACES_RECEIVE_SUCCESS = 'NAMESPACES_RECEIVE_SUCCESS'

export const NAV_MODIFY = 'NAV_MODIFY'
export const NAV_RECEIVE_SUCCESS = 'NAV_RECEIVE_SUCCESS'

export const POST_RECEIVE_FAILURE = 'POST_RECEIVE_FAILURE'
export const POST_RECEIVE_SUCCESS = 'POST_RECEIVE_SUCCESS'
export const POST_REQUEST = 'POST_REQUEST'
export const PUT_RECEIVE_FAILURE = 'PUT_RECEIVE_FAILURE'
export const PUT_RECEIVE_SUCCESS = 'PUT_RECEIVE_SUCCESS'
export const PUT_REQUEST = 'PUT_REQUEST'

export const REPO_FETCH_REQUEST_SUCCESS = 'REPO_FETCH_REQUEST_SUCCESS'

export const RESOURCES_FETCH_REQUEST_LOADING = 'RESOURCE_FETCH_REQUEST_LOADING'
export const RESOURCES_FETCH_REQUEST_SUCCESS = 'RESOURCES_FETCH_REQUEST_SUCCESS'
export const RESOURCE_ADD = 'RESOURCE_ADD'
export const RESOURCE_DELETE= 'RESOURCE_DELETE'
export const RESOURCE_MODIFY = 'RESOURCE_MODIFY'
export const RESOURCE_RECEIVE_FAILURE = 'RESOURCE_RECEIVE_FAILURE'
export const RESOURCE_RECEIVE_SUCCESS = 'RESOURCE_RECEIVE_SUCCESS'
export const RESOURCE_REQUEST = 'RESOURCE_REQUEST'
export const RESOURCE_RESET= 'RESOURCE_RESET'


export const DASHBOARD_RECEIVE_SUCCESS = 'DASHBOARD_RECEIVE_SUCCESS'
export const DASHBOARD_RECEIVE_FAILURE = 'DASHBOARD_RECEIVE_FAILURE'


export const ROLE_RECEIVE_FAILURE = 'ROLE_RECEIVE_FAILURE'
export const ROLE_RECEIVE_SUCCESS = 'ROLE_RECEIVE_SUCCESS'

export const SECONDARY_HEADER_UPDATE = 'SECONDARY_HEADER_UPDATE'

export const SWITCH_NAMESPACE = 'SWITCH_NAMESPACE'

export const TABLE_PAGE_CHANGE = 'TABLE_PAGE_CHANGE'
export const TABLE_SEARCH = 'TABLE_SEARCH'
export const TABLE_SORT = 'TABLE_SORT'

export const TOPOLOGY_FILTERS_UPDATE = 'TOPOLOGY_FILTERS_UPDATE'
export const TOPOLOGY_FILTERS_REQUEST = 'TOPOLOGY_FILTERS_REQUEST'
export const TOPOLOGY_FILTERS_RECEIVE_ERROR = 'TOPOLOGY_FILTERS_RECEIVE_ERROR'
export const TOPOLOGY_FILTERS_RECEIVE_SUCCESS = 'TOPOLOGY_FILTERS_RECEIVE_SUCCESS'
export const TOPOLOGY_SELECTION_UPDATE = 'TOPOLOGY_SELECTION_UPDATE'

export const UICONFIG_RECEIVE_SUCCESS = 'UICONFIG_RECEIVE_SUCCESS'

export const UPDATE_USERPREFERENCES = 'UPDATE_USERPREFERENCES'

export const USERPREFERENCES_RECEIVE_FAILURE = 'USERPRFERENCES_RECEIVE_FAILURE'
export const USERPREFERENCES_RECEIVE_SUCCESS = 'USERPRFERENCES_RECEIVE_SUCCESS'

export const USER_LOGIN_RECEIVE_SUCCESS = 'USER_LOGIN_RECEIVE_SUCCESS'
export const USER_LOGOUT_RECEIVE_FAILURE = 'USER_LOGOUT_RECEIVE_FAILURE'
export const USER_LOGOUT_RECEIVE_SUCCESS = 'USER_LOGOUT_RECEIVE_SUCCESS'
export const USER_LOGOUT_REQUEST = 'USER_LOGOUT_REQUEST'
/*
 * other constants specific to particular actions
 */

export const VisibilityFilters = {
  SHOW_ALL: 'SHOW_ALL',
  SHOW_COMPLETED: 'SHOW_COMPLETED',
  SHOW_ACTIVE: 'SHOW_ACTIVE'
}

export const USER_LOGIN_STATUS = {
  LOGGED_IN: 'LOGGED_IN',
  LOGGED_OUT: 'LOGGED_OUT'
}

export const REQUEST_STATUS = {
  INCEPTION: 'INCEPTION',
  DONE: 'DONE',
  IN_PROGRESS: 'IN_PROGRESS',
  ERROR: 'ERROR'
}

export const PAGE_SIZES = {
  DEFAULT: 20,
  VALUES: [5, 10, 20, 50, 75, 100]
}

export const SORT_DIRECTION_ASCENDING = 'asc'
export const SORT_DIRECTION_DESCENDING = 'desc'
