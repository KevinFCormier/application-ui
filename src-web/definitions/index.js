/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import lodash from 'lodash'

import { RESOURCE_TYPES } from '../../lib/shared/constants'

import hcmpods from './hcm-pods'
import hcmnodes from './hcm-nodes'
import hcmpvs from './hcm-pvs'

const resourceData = {
  [RESOURCE_TYPES.HCM_PODS.name]: hcmpods,
  [RESOURCE_TYPES.HCM_NODES.name]: hcmnodes,
  [RESOURCE_TYPES.HCM_PVS.name]: hcmpvs
}

function getResourceData(resourceType) {
  var def = resourceData[resourceType.name]
  if (!def)
    //eslint-disable-next-line no-console
    console.error(`No resource data found for '${resourceType}'`)
  return def
}

export default getResourceData

export function getPrimaryKey(resourceType) {
  let def = getResourceData(resourceType)
  let pk = def.primaryKey

  if (!pk)
    pk = 'metadata.uid'

  return pk
}

export function getURIKey(resourceType) {
  let def = getResourceData(resourceType)
  let uriKey = def.uriKey

  if (!uriKey)
    uriKey = 'metadata.name'

  return uriKey
}

export function getDefaultSearchField(resourceType) {
  var def = getResourceData(resourceType)
  let sf = def && def.defaultSearchField
  if (!def || !def.tableKeys || (def && def.tableKeys.length < 1))
    //eslint-disable-next-line no-console
    console.error(`No table keys found in ${resourceType} resource definition`)
  if (!sf)
    sf = def && def.tableKeys && def.tableKeys[0] && def.tableKeys[0].resourceKey
  return sf
}

export function getDefaultSortField(resourceType) {
  var def = getResourceData(resourceType)
  let sf = def && def.defaultSortField
  if (!def || !def.tableKeys || (def && def.tableKeys.length < 1))
    //eslint-disable-next-line no-console
    console.error(`No table keys found in ${resourceType} resource definition`)
  if (!sf) {
    sf = def && def.tableKeys && def.tableKeys[0] && def.tableKeys[0].resourceKey
  }
  if (!sf)
    //eslint-disable-next-line no-console
    console.error(`No sortable fields defined for '${resourceType}' resource definition`)
  return sf
}

export function getLink(link, resource) {
  if(typeof(link) === 'boolean') {
    return `/${resource.metadata.namespace}/${resource.metadata.name}`
  } else if (typeof(link) === 'function') {
    return link(resource)
  } else {
    const parts = link.split('/')
    let path = ''
    parts.forEach(part => {
      const value = lodash.get(resource, part)
      path += `/${encodeURIComponent(value)}`
    })
    return path
  }
}
