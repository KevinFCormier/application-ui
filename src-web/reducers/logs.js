/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/

import {
  LOGS_RECEIVE_SUCCESS,
  LOGS_RECEIVE_FAILURE,
  LOGS_RECEIVE_IN_PROGRESS,
  LOGS_RESET,
  REQUEST_STATUS
} from '../actions'

export const logs = (state = null, action) => {
  switch (action.type) {
  case LOGS_RESET:
    return { logs: null, status: REQUEST_STATUS.DONE }
  case LOGS_RECEIVE_IN_PROGRESS:
    return { ...state, status: LOGS_RECEIVE_IN_PROGRESS }
  case LOGS_RECEIVE_SUCCESS:
    return { ...action, status: REQUEST_STATUS.DONE }
  case LOGS_RECEIVE_FAILURE:
    return {
      ...state,
      status: REQUEST_STATUS.ERROR,
      errorMessage: action.err && action.err.details,
      statusCode: action.err && action.err.statusCode
    }
  default:
    return state
  }
}
