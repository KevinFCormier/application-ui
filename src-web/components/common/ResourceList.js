/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/

'use strict'

import React from 'react'
import ResourceTable from './ResourceTable'
import { REQUEST_STATUS } from '../../actions/index'
import NoResource from './NoResource'
import { connect } from 'react-redux'
import { changeTablePage, searchTable, sortTable, fetchResources, updateSecondaryHeader } from '../../actions/common'
import TableHelper from '../../util/table-helper'
import { Loading, Notification } from 'carbon-components-react'
import { withRouter } from 'react-router-dom'
import msgs from '../../../nls/platform.properties'
import headerMsgs from '../../../nls/header.properties'
import config from '../../../lib/shared/config'

class ResourceList extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      xhrPoll: false
    }
  }

  componentWillMount() {
    const { updateSecondaryHeader, tabs, title } = this.props
    updateSecondaryHeader(headerMsgs.get(title, this.context.locale), tabs)
    if (parseInt(config['featureFlags:liveUpdates']) === 2) {
      var intervalId = setInterval(this.reload.bind(this), config['featureFlags:liveUpdatesPollInterval'])
      this.setState({ intervalId: intervalId })
    }
  }

  componentDidMount() {
    const { fetchResources } = this.props
    fetchResources()
  }

  reload() {
    if (this.props.status === REQUEST_STATUS.DONE) {
      this.setState({ xhrPoll: true })
      this.props.fetchResources()
    }
  }

  componentWillUnmount() {
    clearInterval(this.state.intervalId)
  }

  render() {
    const {
      items,
      itemIds,
      page,
      pageSize,
      sortDirection,
      totalFilteredItems,
      status,
      sortTable,
      sortColumn,
      changeTablePage,
      searchTable,
      staticResourceData,
      searchValue,
      resourceType,
      statusCode,
      children,
      namespace
    } = this.props
    const { locale } = this.context

    if (status === REQUEST_STATUS.ERROR && !this.state.xhrPoll) {
      return <Notification
        title=''
        className='persistent'
        subtitle={msgs.get(`error.${(statusCode === 401 || statusCode === 403) ? 'unauthorized' : 'default'}.description`, locale)}
        kind='error' />
    }

    if (status !== REQUEST_STATUS.DONE && !this.state.xhrPoll)
      return <Loading withOverlay={false} className='content-spinner' />

    const actions = React.Children.map(children, action => {
      if (action.props.disabled)
        return null
      return React.cloneElement(action, { resourceType })
    })

    if (items || searchValue)
      return <ResourceTable
        actions={actions}
        staticResourceData={staticResourceData}
        page={page}
        pageSize={pageSize}
        itemIds={itemIds}
        sortDirection={sortDirection}
        sortColumn={sortColumn}
        status={status}
        items={items}
        totalFilteredItems={totalFilteredItems}
        resourceType={resourceType}
        changeTablePage={changeTablePage}
        handleSort={TableHelper.handleSort.bind(this, sortDirection, sortColumn, sortTable)}
        handleSearch={TableHelper.handleInputValue.bind(this, searchTable)}
        searchValue={searchValue}
        tableActions={staticResourceData.tableActions} />

    const resourceName = msgs.get('no-resource.' + resourceType.name.toLowerCase(), locale)
    return (
      <NoResource
        title={msgs.get('no-resource.title', [resourceName], locale)}
        detail={msgs.get(`no-resource.detail${namespace.split(',').length > 1 ? '' : '-namespace'}`, [resourceName, namespace], locale)}>
        {actions}
      </NoResource>
    )
  }

  handleResourceAddedEvent(event) {
    this.props.addResource(event)
  }

  handleResourceModifiedEvent(event) {
    this.props.modifyResource(event)
  }

  handleResourceDeletedEvent(event) {
    this.props.deleteResource(event)
  }
}

const mapStateToProps = (state, ownProps) => {
  const { list: typeListName } = ownProps.resourceType,
        error = state[typeListName].err,
        visibleResources = ownProps.getVisibleResources(state, {'storeRoot': typeListName})
  return {
    items: visibleResources.normalizedItems,
    itemIds: visibleResources.items,
    totalFilteredItems: visibleResources.totalResults,
    totalPages: visibleResources.totalPages,
    status: state[typeListName].status,
    page: state[typeListName].page,
    pageSize: state[typeListName].itemsPerPage,
    sortDirection: state[typeListName].sortDirection,
    sortColumn: state[typeListName].sortColumn,
    searchValue: state[typeListName].search,
    statusCode: error && error.response && error.response.status,
  }
}

const mapDispatchToProps = (dispatch, ownProps) => {
  const resourceType = ownProps.resourceType
  return {
    fetchResources: () => dispatch(fetchResources(resourceType)),
    changeTablePage: page => dispatch(changeTablePage(page, resourceType)),
    searchTable: search => dispatch(searchTable(search, resourceType)),
    sortTable: (sortDirection, sortColumn) => dispatch(sortTable(sortDirection, sortColumn, resourceType)),
    updateSecondaryHeader: (title, tabs) => dispatch(updateSecondaryHeader(title, tabs)),
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ResourceList))
