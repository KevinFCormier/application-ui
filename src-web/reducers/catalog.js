/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/

import * as Actions from '../actions/index'
// if term already exists in termsList, remove it. otherwise, add it.
// only works for value types, not arrays or objects.
export const addOrRemove = (termsList = [], term) => {
  return termsList.includes(term)
    ? termsList.filter(el => el !== term)
    : [...termsList, term]
}

const initialStateCatalog = {
  items: [],
  repos: [],
  selection: {
    repoName: '',
    url: '',
    name: '',
  },
  catalogFetchFailure: false,
  catalogFetchLoading: false,
  catalogInstallValidationFailure: false,
  catalogInstallFailure: false,
  catalogInstallLoading: false,
  filters: {
    searchText: '',
  },
  dropdownFiltersVisibility: false,
}

const catalog = (state = initialStateCatalog, action) => {
  switch (action.type) {
  case Actions.CATALOG_FETCH_ERROR_STATUS_CHANGE: {
    const { status = !state.catalogFetchFailure } = action.payload
    return {
      ...state,
      catalogFetchFailure: status,
    }
  }
  case Actions.CATALOG_RESOURCE_SELECT: {
    const { name, url, repoName } = action.payload
    return { ...state, selection: { name, url, repoName } }
  }
  case Actions.RESOURCES_FETCH_REQUEST_LOADING: {
    const { status = !state.catalogFetchLoading } = action.payload
    return { ...state, catalogFetchLoading: status }
  }
  case Actions.CATALOG_INSTALL_LOADING: {
    const { status } = action.payload
    return { ...state, catalogInstallLoading: status }
  }
  case Actions.CATALOG_INSTALL_FAILURE: {
    const { status } = action.payload
    return { ...state, catalogInstallFailure: status }
  }
  case Actions.CATALOG_INSTALL_VALIDATION_FAILURE: {
    const { status } = action.payload
    return { ...state, catalogInstallValidationFailure: status }
  }
  case Actions.RESOURCES_FETCH_REQUEST_SUCCESS: {
    const { items } = action.payload
    return {
      ...state,
      catalogFetchFailure: false,
      items,
    }
  }
  case Actions.CATALOG_RESOURCE_FILTER_SEARCH: {
    const { searchText } = action.payload
    return {
      ...state,
      filters: { ...state.filters, searchText },
    }
  }
  case Actions.REPO_FETCH_REQUEST_SUCCESS: {
    const { repos } = action.payload
    return { ...state, repos }
  }
  case Actions.CATALOG_DROPDOWN_FILTERS_VISIBILITY_TOGGLE: {
    const { visibility } = action.payload
    const visibilityWithDefault = // if undefined or not a boolean, toggle.
        typeof visibility !== 'boolean'
          ? !state.dropdownFiltersVisibility
          : visibility
    return { ...state, dropdownFiltersVisibility: visibilityWithDefault }
  }
  default:
    return state
  }
}

export default catalog
