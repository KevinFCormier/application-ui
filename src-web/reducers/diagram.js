/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/

import * as Actions from '../actions'
import { getFilterState, saveFilterState } from '../../lib/client/filter-helper'

const initialState = {
  diagramFilters: [],
}

export const diagram = (state=initialState, action) => {
  switch (action.type){
  default:
  case '@@INIT':{
    return initialState
  }
  case Actions.DIAGRAM_RESTORE_FILTERS: {
    const { namespace, name, initialDiagramFilters } = action
    const {filters: diagramFilters} = getFilterState(initialDiagramFilters, namespace, name)
    state.diagramFilters = diagramFilters
    return {...state}
  }
  case Actions.DIAGRAM_SAVE_FILTERS: {
    const { namespace, name, diagramFilters } = action
    saveFilterState(namespace, name, {filters:diagramFilters})
    state.diagramFilters = diagramFilters
    return {...state}
  }
  }
}
