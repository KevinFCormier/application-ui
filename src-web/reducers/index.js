/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/*
 *
 * Things you should never do inside a reducer:
 *
 * - Mutate its arguments
 * - Perform side effects like API calls and routing transitions
 * - Call non-pure functions, e.g. Date.now() or Math.random()
 *
 * Reducers must be deterministic pure functions.  Given the same arguments, it should calculate the next state and return it.
 * No surprises. No side effects. No API calls. No mutations. Just a calculation.
 *
 * Selectors should sit along side reducers.
 */

import { createResourceReducer, resourceReducerFunction } from './common'
import { RESOURCE_TYPES } from '../../lib/shared/constants'
import lodash from 'lodash'

export { user, loggedIn } from './user'
export { secondaryHeader } from './common'

export { modal } from './modal'

//the exported function name must match the resourceType value
export const HCMApplicationList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_APPLICATIONS))
export const HCMClusterList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_CLUSTERS))
export const HCMComplianceList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_COMPLIANCES))
export const HCMDashboardList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_DASHBOARD))
export const HCMNodeList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_NODES))
export const HCMPersistentVolumeList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_PVS))
export const HCMPersistentVolumeClaimList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_PVS_CLAIM))
export const HCMPodList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_PODS))
export const HCMPolicyList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_POLICIES))
export const HCMReleaseList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_RELEASES))
export const HCMRepositoryList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_REPOSITORIES))

export { default as catalog } from './catalog'
export { topology } from './topology'
export { resourceFilters } from './filter'

function predicate(resourceType, action) {
  if (lodash.isEqual(resourceType, action.resourceType))
    return true
  return lodash.find(lodash.values(resourceType), type => {
    if (typeof type === 'string') {
      return type.indexOf(action.resourceType) > -1
    }
    return false
  })
}
