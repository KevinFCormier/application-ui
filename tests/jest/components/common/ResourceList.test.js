/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
const React = require("../../../../node_modules/react");

import ResourceList from "../../../../src-web/components/common/ResourceList";

import renderer from "react-test-renderer";
import * as reducers from "../../../../src-web/reducers";

import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import thunkMiddleware from "redux-thunk";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

const preloadedState = window.__PRELOADED_STATE__;
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const middleware = [thunkMiddleware];

const store = createStore(
  combineReducers(reducers),
  preloadedState,
  composeEnhancers(applyMiddleware(...middleware))
);

describe("ResourceList", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <ResourceList {...props} />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const props = {
  secondaryHeaderProps: { title: "routes.applications" },
  match: {
    path: "/multicloud/applications",
    url: "/multicloud/applications",
    isExact: true,
    params: {}
  },
  location: {
    pathname: "/multicloud/applications",
    search: "",
    hash: "",
    state: undefined,
    key: "4wnt09"
  },
  history: { length: 16, action: "PUSH", location: {} },
  staticContext: undefined,
  detailsTabs: [],
  routes: [],
  resourceType: { name: "QueryApplications", list: "QueryApplicationList" },
  staticResourceData: {
    defaultSortField: "name",
    uriKey: "name",
    primaryKey: "name",
    secondaryKey: "namespace"
  },
  getVisibleResources: function getVisibleResources(state, obj) {
    return { normalizedItems: ["item1", "item2"] };
  },
  buttons: [{}],
  modules: [{}],
  tableTitle: "All applications",
  tableName: "All applications",
  tabs: undefined,
  title: "routes.applications",
  children: [{}]
};
