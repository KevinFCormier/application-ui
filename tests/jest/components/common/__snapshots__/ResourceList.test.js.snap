// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceList renders correctly 1`] = `
<div
  className="bx--module bx--module--single bx--tile search-query-card-loading-apps "
>
  <div
    className="bx--module__inner"
  >
    <div
      className="search-query-card-loading-apps "
    >
      <p
        className="bx--skeleton__text"
        style={
          Object {
            "width": "100%",
          }
        }
      />
      <p
        className="bx--skeleton__text"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
